server/scsynth/SC_World.cpp

```diff
@@ -150,16 +157,14 @@ void zfree(void* ptr) { return free_alig(ptr); }
 // Set denormal FTZ mode on CPUs that need/support it.
 void sc_SetDenormalFlags();
 
-#ifdef __SSE2__
-#    include <xmmintrin.h>
+#if defined(BOOST_ARCH_X86_64) || (BOOST_HW_SIMD_X86 >= BOOST_HW_SIMD_X86_SSE2_VERSION)
 
 void sc_SetDenormalFlags() {
     _MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_ON);
     _mm_setcsr(_mm_getcsr() | 0x40); // DAZ
 }
 
-#elif defined(__SSE__)
-#    include <xmmintrin.h>
+#elif BOOST_HW_SIMD_X86 >= BOOST_X86_SSE_VERSION
```

Comment:
did some looking around -- this code can be changed to match the supernova implementation below. all the operations in it are available and work in the SSE instruction set. in fact the entire thing can be reduced to

```cpp
static void sc_SetDenormalFlags() {
#if BOOST_HW_SIMD_X86 >= BOOST_HW_SIMD_X86_SSE_VERSION
    _MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_ON);
    _mm_setcsr(_mm_getcsr() | 0x40); // DAZ
#endif
}
```

Reply:
you can also then remove the `architecture.h` include above